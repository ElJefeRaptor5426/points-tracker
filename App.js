import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import PointsDisplay from './components/PointsDisplay';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      points: 0,
    };
  } 
  render() {
    return (
      <View style={styles.container}>
        <Text>Points:</Text>
        <PointsDisplay points={this.state.points} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
