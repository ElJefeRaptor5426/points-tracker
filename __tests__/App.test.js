import {shallow, render} from 'enzyme';
import App from '../App';
import React from 'react';
import renderer from 'react-test-renderer';

describe('App component', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(<App />)
    expect(wrapper.toJSON()).toMatchSnapshot();
  })
})

