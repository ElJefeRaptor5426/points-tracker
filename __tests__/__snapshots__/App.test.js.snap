// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 50,
        "padding": 15,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
          -
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "color": "#000",
            "fontSize": 42,
            "textAlign": "center",
          }
        }
      >
        <Text>
          Points:
        </Text>
        <Text>
          0
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "green",
              },
            ]
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 500,
        "maxHeight": 500,
        "padding": 50,
        "width": "100%",
      }
    }
  >
    <View
      ListHeaderComponent={[Function]}
      data={Array []}
      keyExtractor={[Function]}
      renderItem={[Function]}
      scrollEnabled={true}
    />
  </View>
</View>
`;
